Sources/ContextDataCapturer.swift

```diff
@@ -0,0 +1,322 @@
+    public func startCapture() {
+        contextDataCaptureIsActive = true
+    }
+
+    /// Adds the Event to the capture list, given a context data capturing session is active; use `startCapture()`to
+    /// start a capture session
+    public func addEvent(_ event: Event) {
+        if contextDataCaptureIsActive {
+            contextDataStore.append(event)
+        }
+    }
+
+    /// Stops a debug context data capture session and prints a report of the captured context data keys and values during
+    /// the session to the device logs. The report can then be uploaded to Data Prep for Data Collection to map the context
+    /// data to XDM.
+    ///
+    /// Note, this API is intended for use during debugging and should not be used in a production environment. The SDK log
+    /// level must be configured to `LogLevel.debug` or `LogLevel.trace` for the report to be printed.
+    ///
+    /// - Parameters:
+    ///     - withMerge: Controls if merge logic is applied to captured `Event`s
+    ///     - isMergeCaseSensitive: Controls if merge logic for matching keys uses case sensitive compare or not
+    public func stopCapture(withMerge: Bool, isMergeCaseSensitive: Bool) {
+        contextDataCaptureIsActive = false
+        outputCapturedContextData(withMerge: withMerge, isMergeCaseSensitive: isMergeCaseSensitive)
+        contextDataStore.removeAll()
+    }
+
+    /// Outputs context data that has been captured up to the point the method is called; does not affect capture status or collected data.
+    ///
+    /// - Parameters:
+    ///     - withMerge: Controls if merge logic is applied to captured `Event`s
+    ///     - isMergeCaseSensitive: Controls if merge logic for matching keys uses case sensitive compare or not
+    private func outputCapturedContextData(withMerge: Bool, isMergeCaseSensitive: Bool) {
+        var mergeReport = ""
+
+        func addToMergeReport(text: String) {
+            mergeReport += text + "\n"
+        }
+        if withMerge {
+            guard let firstEvent = contextDataStore.first else {
+                Log.debug(label: EdgeBridgeConstants.LOG_TAG, "No events to merge.")
+                return
+            }
+            // Initialize merge result using first valid event
+            var mergeResult = MergeResult(event: firstEvent, isCaseSensitive: isMergeCaseSensitive, keySetResult: extractKeySet(event: firstEvent, isCaseSensitive: isMergeCaseSensitive))
+            // Start merging next events into result
+            if contextDataStore.count > 1 {
+                for i in 1..<contextDataStore.count {
+                    mergeResult = mergeEvents(mergeResult: mergeResult, eventToMerge: contextDataStore[i])
+                }
+            }
+            addToMergeReport(text: "============== Merge Result =================")
+            addToMergeReport(text: "-------------- Value type/key merge conflicts --------------")
+            for (key, value) in mergeResult.keySet {
```

Comment:
Can we add this logic in a separate `function getMergeConflictReport(mergeResult)`

Reply:
Yes I was thinking to move out the type/key merge conflicts logic.